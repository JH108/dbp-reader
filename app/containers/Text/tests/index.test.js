// import React from 'react';
// import { shallow, mount } from 'enzyme';
// import { fromJS } from 'immutable';
// import Text from '../index';
// import jsonProps from './highlightFormattedTextProps.json';
// import jsonState from './highlightFormattedTextState.json';

// const textProps = {
//   text: [], // array
//   userNotes: [], // array
//   bookmarks: [], // array
//   highlights: [], // array
//   userSettings: fromJS({}), // object
//   formattedSource: fromJS({}), // object
//   nextChapter: () => {}, // func
//   prevChapter: () => {}, // func
//   addBookmark: () => {}, // func
//   addHighlight: () => {}, // func
//   goToFullChapter: () => {}, // func
//   toggleNotesModal: () => {}, // func
//   setActiveNotesView: () => {}, // func
//   toggleInformationModal: () => {}, // func
//   setActiveNote: () => {}, // func
//   activeChapter: 1, // number
//   notesActive: true, // bool
//   invalidBibleId: true, // bool
//   informationActive: true, // bool
//   userAuthenticated: true, // bool
//   loadingNewChapterText: true, // bool
//   userId: '', // string
//   bibleId: '', // string
//   verseNumber: '', // string
//   activeBookId: '', // string
//   activeBookName: '', // string
// };

// xdescribe('<Text />', () => {
//   it('Expect to have unit tests specified', () => {
//     expect(true).toEqual(true);
//   });
//   describe('addHighlight', () => {
//     let wrapper;

//     beforeEach(() => {
//       wrapper = shallow(<Text {...textProps} />);
//       wrapper.instance().state = {
//         contextMenuState: false,
//         footnoteState: false,
//         coords: {},
//         selectedText: '',
//         firstVerse: 0,
//         lastVerse: 0,
//         highlightActive: textProps.highlights || false,
//         handlersAreSet: false,
//       };
//     });
//     it('Should be defined', () => {
//       expect(wrapper.instance().addHighlight).toBeDefined();
//     });
//     it('Should be a function', () => {
//       expect(typeof wrapper.instance().addHighlight === 'function').toEqual(
//         true,
//       );
//     });
//     xit('Should create a highlight object for plain text left to right in same verse', () => {
//       expect(true).toEqual(false);
//     });
//     xit('Should create a highlight object for plain text left to right spanning two verses', () => {
//       expect(true).toEqual(false);
//     });
//     xit('Should create a highlight object for plain text left to right in same verse that overlaps another highlight', () => {
//       expect(true).toEqual(false);
//     });
//     xit('Should create a highlight object for plain text left to right in same verse where start is not on a verse', () => {
//       expect(true).toEqual(false);
//     });
//     xit('Should create a highlight object for plain text left to right spanning two verses and that overlaps another highlight', () => {
//       expect(true).toEqual(false);
//     });
//     xit('Should create a highlight object for formatted text left to right in same verse', () => {
//       expect(true).toEqual(false);
//     });
//     xit('Should create a highlight object for formatted text left to right spanning two verses', () => {
//       expect(true).toEqual(false);
//     });
//     xit('Should create a highlight object for formatted text left to right in same verse that overlaps another highlight', () => {
//       expect(true).toEqual(false);
//     });
//     it('Should create a highlight object for formatted text left to right that has another non-overlapping highlight in same verse', () => {
//       const mountWrap = mount(
//         <Text
//           {...jsonProps.verseHasHighlight}
//           formattedSource={fromJS(jsonProps.verseHasHighlight.formattedSource)}
//           userSettings={fromJS(jsonProps.verseHasHighlight.userSettings)}
//         />,
//       );
//       const expectedObject = {
//         chapter: 6,
//         verse_start: 7,
//         highlight_start: 40,
//         highlighted_words: 6,
//         highlighted_color: '#86A',
//         bible_id: 'ENGWEB',
//         book_id: 'MAT',
//       };
//       mountWrap.setState({ ...jsonState.verseHasHighlight });
//       // mountWrap.update();
//       // console.log(mountWrap.find('main').children('.v.MAT6_6'));
//       // Gets the rendered element
//       // console.log(mountWrap.find('main').children().forEach((c) => console.log('child exists', c.render())));
//       // console.log(mountWrap.find('.wj'));
//       // console.log(mountWrap.find('arrow-wrapper'));
//       // expect(mountWrap.instance().addHighlight({ color: '#000', popupCoords: { x: 0, y: 0 } })).toEqual(expectedObject);
//       expect(true).toBe(expectedObject);
//       mountWrap.unmount();
//     });
//     xit('Should create a highlight object for formatted text left to right in same verse that overlaps a footnote', () => {
//       expect(true).toEqual(false);
//     });
//     xit('Should create a highlight object for formatted text left to right in same verse that overlaps a note', () => {
//       expect(true).toEqual(false);
//     });
//     xit('Should create a highlight object for formatted text left to right in same verse that overlaps a bookmark', () => {
//       expect(true).toEqual(false);
//     });
//     xit('Should create a highlight object for formatted text left to right in same verse that overlaps words of Jesus', () => {
//       expect(true).toEqual(false);
//     });
//     xit('Should create a highlight object for formatted text left to right in same verse that overlaps another element', () => {
//       expect(true).toEqual(false);
//     });
//     xit('Should create a highlight object for formatted text left to right in same verse where start is not on a verse', () => {
//       expect(true).toEqual(false);
//     });
//     xit('Should create a highlight object for formatted text left to right spanning two verses and overlaps another highlight', () => {
//       expect(true).toEqual(false);
//     });
//     xit('Should create a highlight object for formatted text left to right spanning two verses and overlaps a footnote', () => {
//       expect(true).toEqual(false);
//     });
//     xit('Should create a highlight object for formatted text left to right spanning two verses and overlaps a note', () => {
//       expect(true).toEqual(false);
//     });
//     xit('Should create a highlight object for formatted text left to right spanning two verses and overlaps a bookmark', () => {
//       expect(true).toEqual(false);
//     });
//     xit('Should create a highlight object for formatted text left to right spanning two verses and overlaps words of Jesus', () => {
//       expect(true).toEqual(false);
//     });
//     xit('Should create a highlight object for formatted text left to right spanning two verses and overlaps another element', () => {
//       expect(true).toEqual(false);
//     });
//     xit('Should create a highlight object for formatted text left to right spanning two verses where both verses are split into quotes', () => {
//       expect(true).toEqual(false);
//     });
//     xit('Should create a highlight object for formatted text left to right in the same verse where the verse is split into quotes', () => {
//       expect(true).toEqual(false);
//     });
//     xit('Should create a highlight object for plain text right to left in same verse', () => {
//       expect(true).toEqual(false);
//     });
//     xit('Should create a highlight object for plain text right to left spanning two verses', () => {
//       expect(true).toEqual(false);
//     });
//     xit('Should create a highlight object for plain text right to left in same verse that overlaps another highlight', () => {
//       expect(true).toEqual(false);
//     });
//     xit('Should create a highlight object for plain text right to left in same verse where start is not on a verse', () => {
//       expect(true).toEqual(false);
//     });
//     xit('Should create a highlight object for plain text right to left spanning two verses and that overlaps another highlight', () => {
//       expect(true).toEqual(false);
//     });
//     xit('Should create a highlight object for formatted text right to left in same verse', () => {
//       expect(true).toEqual(false);
//     });
//     xit('Should create a highlight object for formatted text right to left spanning two verses', () => {
//       expect(true).toEqual(false);
//     });
//     xit('Should create a highlight object for formatted text right to left in same verse that overlaps another highlight', () => {
//       expect(true).toEqual(false);
//     });
//     xit('Should create a highlight object for formatted text right to left in same verse that overlaps a footnote', () => {
//       expect(true).toEqual(false);
//     });
//     xit('Should create a highlight object for formatted text right to left in same verse that overlaps a note', () => {
//       expect(true).toEqual(false);
//     });
//     xit('Should create a highlight object for formatted text right to left in same verse that overlaps a bookmark', () => {
//       expect(true).toEqual(false);
//     });
//     xit('Should create a highlight object for formatted text right to left in same verse that overlaps words of Jesus', () => {
//       expect(true).toEqual(false);
//     });
//     xit('Should create a highlight object for formatted text right to left in same verse that overlaps another element', () => {
//       expect(true).toEqual(false);
//     });
//     xit('Should create a highlight object for formatted text right to left in same verse where start is not on a verse', () => {
//       expect(true).toEqual(false);
//     });
//     xit('Should create a highlight object for formatted text right to left spanning two verses and overlaps another highlight', () => {
//       expect(true).toEqual(false);
//     });
//     xit('Should create a highlight object for formatted text right to left spanning two verses and overlaps a footnote', () => {
//       expect(true).toEqual(false);
//     });
//     xit('Should create a highlight object for formatted text right to left spanning two verses and overlaps a note', () => {
//       expect(true).toEqual(false);
//     });
//     xit('Should create a highlight object for formatted text right to left spanning two verses and overlaps a bookmark', () => {
//       expect(true).toEqual(false);
//     });
//     xit('Should create a highlight object for formatted text right to left spanning two verses and overlaps words of Jesus', () => {
//       expect(true).toEqual(false);
//     });
//     xit('Should create a highlight object for formatted text right to left spanning two verses and overlaps another element', () => {
//       expect(true).toEqual(false);
//     });
//     xit('Should create a highlight object for formatted text right to left spanning two verses where both verses are split into quotes', () => {
//       expect(true).toEqual(false);
//     });
//     xit('Should create a highlight object for formatted text right to left in the same verse where the verse is split into quotes', () => {
//       expect(true).toEqual(false);
//     });
//   });
// });
