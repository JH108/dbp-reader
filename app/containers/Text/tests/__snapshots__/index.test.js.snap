// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> Should match previous snapshot with all options true 1`] = `
<div
  className="text-container scrolled-down video-player-open has-video"
>
  <div
    className="loading-spinner"
  >
    <span
      className="loading-spinner1 loading-spinner-child"
    />
    <span
      className="loading-spinner2 loading-spinner-child"
    />
    <span
      className="loading-spinner3 loading-spinner-child"
    />
    <span
      className="loading-spinner4 loading-spinner-child"
    />
    <span
      className="loading-spinner5 loading-spinner-child"
    />
    <span
      className="loading-spinner6 loading-spinner-child"
    />
    <span
      className="loading-spinner7 loading-spinner-child"
    />
    <span
      className="loading-spinner8 loading-spinner-child"
    />
    <span
      className="loading-spinner9 loading-spinner-child"
    />
    <span
      className="loading-spinner10 loading-spinner-child"
    />
    <span
      className="loading-spinner11 loading-spinner-child"
    />
    <span
      className="loading-spinner12 loading-spinner-child"
    />
  </div>
</div>
`;

exports[`<Text /> Should match previous snapshot with audioPlayerState true 1`] = `
<div
  className="text-container audio-player-open"
  id="text-container-parent"
>
  <div
    className="main-wrapper"
  />
</div>
`;

exports[`<Text /> Should match previous snapshot with changingVersion true 1`] = `
<div
  className="text-container"
>
  <div
    className="loading-spinner"
  >
    <span
      className="loading-spinner1 loading-spinner-child"
    />
    <span
      className="loading-spinner2 loading-spinner-child"
    />
    <span
      className="loading-spinner3 loading-spinner-child"
    />
    <span
      className="loading-spinner4 loading-spinner-child"
    />
    <span
      className="loading-spinner5 loading-spinner-child"
    />
    <span
      className="loading-spinner6 loading-spinner-child"
    />
    <span
      className="loading-spinner7 loading-spinner-child"
    />
    <span
      className="loading-spinner8 loading-spinner-child"
    />
    <span
      className="loading-spinner9 loading-spinner-child"
    />
    <span
      className="loading-spinner10 loading-spinner-child"
    />
    <span
      className="loading-spinner11 loading-spinner-child"
    />
    <span
      className="loading-spinner12 loading-spinner-child"
    />
  </div>
</div>
`;

exports[`<Text /> Should match previous snapshot with chapterTextLoadingState true 1`] = `
<div
  className="text-container"
>
  <div
    className="loading-spinner"
  >
    <span
      className="loading-spinner1 loading-spinner-child"
    />
    <span
      className="loading-spinner2 loading-spinner-child"
    />
    <span
      className="loading-spinner3 loading-spinner-child"
    />
    <span
      className="loading-spinner4 loading-spinner-child"
    />
    <span
      className="loading-spinner5 loading-spinner-child"
    />
    <span
      className="loading-spinner6 loading-spinner-child"
    />
    <span
      className="loading-spinner7 loading-spinner-child"
    />
    <span
      className="loading-spinner8 loading-spinner-child"
    />
    <span
      className="loading-spinner9 loading-spinner-child"
    />
    <span
      className="loading-spinner10 loading-spinner-child"
    />
    <span
      className="loading-spinner11 loading-spinner-child"
    />
    <span
      className="loading-spinner12 loading-spinner-child"
    />
  </div>
</div>
`;

exports[`<Text /> Should match previous snapshot with hasVideo true 1`] = `
<div
  className="text-container has-video"
  id="text-container-parent"
>
  <div
    className="main-wrapper"
  />
</div>
`;

exports[`<Text /> Should match previous snapshot with loadingNewChapterText true 1`] = `
<div
  className="text-container"
>
  <div
    className="loading-spinner"
  >
    <span
      className="loading-spinner1 loading-spinner-child"
    />
    <span
      className="loading-spinner2 loading-spinner-child"
    />
    <span
      className="loading-spinner3 loading-spinner-child"
    />
    <span
      className="loading-spinner4 loading-spinner-child"
    />
    <span
      className="loading-spinner5 loading-spinner-child"
    />
    <span
      className="loading-spinner6 loading-spinner-child"
    />
    <span
      className="loading-spinner7 loading-spinner-child"
    />
    <span
      className="loading-spinner8 loading-spinner-child"
    />
    <span
      className="loading-spinner9 loading-spinner-child"
    />
    <span
      className="loading-spinner10 loading-spinner-child"
    />
    <span
      className="loading-spinner11 loading-spinner-child"
    />
    <span
      className="loading-spinner12 loading-spinner-child"
    />
  </div>
</div>
`;

exports[`<Text /> Should match previous snapshot with menuIsOpen true 1`] = `
<div
  className="text-container"
  id="text-container-parent"
>
  <div
    className="main-wrapper"
  />
</div>
`;

exports[`<Text /> Should match previous snapshot with no options true 1`] = `
<div
  className="text-container"
  id="text-container-parent"
>
  <div
    className="main-wrapper"
  />
</div>
`;

exports[`<Text /> Should match previous snapshot with subFooterOpen true 1`] = `
<div
  className="text-container sub-footer-open"
  id="text-container-parent"
>
  <div
    className="main-wrapper"
  />
</div>
`;

exports[`<Text /> Should match previous snapshot with videoPlayerOpen true 1`] = `
<div
  className="text-container"
  id="text-container-parent"
>
  <div
    className="main-wrapper"
  />
</div>
`;
